app/helpers/katello/hosts_and_hostgroups_helper.rb

```diff
@@ -98,5 +115,37 @@ def content_views_for_host(host, options)
       view_options.insert(0, include_blank) if include_blank
       view_options.html_safe
     end
+
+    def kickstart_repository_options(host)
+      view_options = []
+
+      if host.present? && host.operatingsystem.is_a?(Redhat)
+        new_host = ::Host.new
+        new_host.operatingsystem = host.operatingsystem
+        new_host.architecture = host.architecture
+        new_host.content_facet = Host::ContentFacet.new(:lifecycle_environment_id => host.lifecycle_environment_id,
+                                                        :content_view_id => host.content_view_id)
+        new_host.content_source = host.content_source
+        view_options = new_host.operatingsystem.kickstart_repos(new_host).map { |repo| OpenStruct.new(repo) }
+      elsif ["operatingsystem_id", 'content_view_id',
+             'lifecycle_environment_id', 'content_source_id',
+             'architecture_id'].all? { |key| !params[key].blank? }
```

Comment:
So this method gets called in 2 places

1. On initial page load or a host group selection. At that point the host object is already populated and we should just use that.
1. Once you chose a diff os/content source/arch/lifecycle env/cv

In case  2 we want it to play by the rules of "one of these params" and in case 1 we want it to behave as if everything is already set right and we just need the KS repos ..